### ISSUE TITLE
Updating to BP 2+ (filter not being applied)

### ISSUE THREAD

**Issue by BoweFrankema** [boonebgorges/bp-better-directories#6]
Hiya Boone,

I've been trying to update this plugin for use with more recent version of BuddyPress. I've made a few changes to the javascript is enqueued:

```
    function enqueue_scripts() {
        if ( bp_is_directory() && bp_is_members_component() ) {
            wp_enqueue_script( 'jquery-loadmask', BPBD_INSTALL_URL . '/includes/lib/jquery.loadmask/jquery.loadmask.min.js', array( 'jquery' ) );
            wp_enqueue_script( 'bpbd-js', BPBD_INSTALL_URL . '/includes/js/bpbd.js', array( 'bp-legacy-js', 'jquery-loadmask') );
        }
```

and made the BPBD.js less specific (replacing #content with #buddypress). 

But I've not been able to get the filtering to work. I think it's because the search terms are not being passed/applied to the Ajax query. The following error is being given:

On line 277 the following error occurs
Uncaught TypeError: Cannot read property 'get' of undefined

```
  if (jq.query.get('s') && !search_terms)
    search_terms = jq.query.get('s');
```

I _think_ it has something to do with the SQL filter being applied. But I can't be sure. 

https://github.com/boonebgorges/bp-better-directories/blob/master/bpbd.php#L104

It would be cool to get this working but if not, I'll use BP Profile Search plugin :-)

**Comment by boonebgorges** [comment]
@BoweFrankema Yeah, I can take a look. What branch of the plugin are you using?

**Comment by BoweFrankema** [comment]
I'm using the Master branch.. I assumed that was the latest version?
